airflow/cli/__init__.py

```diff
@@ -61,7 +61,7 @@
 click_log_file = click.option(
     "-l",
     "--log-file",
-    type=click.Path(exists=True, dir_okay=False, writable=True),
+    type=click.Path(exists=False, dir_okay=False, writable=True),
```

Comment:
In the current cli, log-file, pid, stdout, stderr are created if they don't exist.